### DISCUSSION TITLE
Livewire 3 Events

### DISCUSSION THREAD

**Discussion by MuhammadSyahmi98** [livewire/livewire#7206] [1. Help]
I'm using Livewire 3's wire:navigate for single page navigation in my application. I have a page with a button that triggers a confirm dialog before deleting a record. This works on first load, but when I navigate to another page and back, the button click no longer triggers the confirm dialog.

The initial code listens for 'livewire:initialized' and attaches the confirm dialog event:

`document.addEventListener('livewire:initialized', () => {
  @this.on('triggerDelete', (event) => {
    // show confirm dialog 
  }) 
})`

However, this stops working after navigating away and back. If I change to listen on the 'triggerDelete' event directly on window, it works consistently:

`window.addEventListener('triggerDelete', event => {
  // show confirm dialog
})`

I can't figure out why the first approach with 'livewire:initialized' works only on first load. It seems like it should attach the listener each time the component initializes.

Has anyone faced a similar issue with Livewire event listeners getting detached on navigation? What is the correct way to attach Livewire event listeners that persist across component/page reloads? Any help is appreciated!"

Let me know if you would like me to modify or expand this question in any way!

**Comment by danie-ramdhani** [comment]
try add `livewire:navigated` event too (https://livewire.laravel.com/docs/navigate#dont-rely-on-domcontentloaded): 

```js
document.addEventListener('livewire:navigated', () => { 
    // ...
})
```
